components/service/glean/src/main/java/mozilla/components/service/glean/CommonMetricData.kt

```diff
@@ -74,4 +74,18 @@ interface CommonMetricData {
         val filteredNames = sendInPings.filter { it != DEFAULT_STORAGE_NAME }
         return filteredNames + defaultStorageDestinations
     }
+
+    /**
+     * Helper function to return the name of the stored metric.
+     * This is used to support empty categories.
+     */
+    fun getStoredName(): String {
```

Comment:
It feels weird to have this function here, as this is an engine implementation detail.

Reply:
I actually disagree.  "Name" of anything should be a property of the object, not it's storage engine.  I renamed this to `getKeyName` to hopefully be less weird and actually refactored out the other `getStoredName` from the `GenericScalarStorageEngine` to illustrate that it's more concise this way.  I don't have strong feelings about going either way, it just made it much more concise to call this in the test functions to get the correct metric by going this way.

Reply:
> 
> 
> I actually disagree. "Name" of anything should be a property of the object, not it's storage engine. 

I agree, in general, with that statement! However the object itself doesn't really care too much about it will be stored: it shouldn't know we'll store `name` vs `category.name`.

> I renamed this to `getKeyName` to hopefully be less weird and actually refactored out the other `getStoredName` from the `GenericScalarStorageEngine` to illustrate that it's more concise this way. I don't have strong feelings about going either way, I just think it just made calling it much more concise in the test functions in each of the metric type definitions.

This is actually much better: my main point was having something related to "store" in the `CommonMetric` :) Now, since we're basically using this as an id, what if we call this `getIdentifier`?

nit: we could probably make this a property with a custom accessor, maybe also `internal` only so that it doesn't show up in the IDE:

```
val identifier: String
    get() = if (category.isEmpty()) name else "$category.$name"
```

Reply:
+1 I like this even better!